### PR TITLE
STORM-3945 - Upgrade Zookeeper to 3.9.0 & Curator 5.5.0

### PR DESCRIPTION
**Pull request by rzo1**
## What is the purpose of the change

- Replace EOL Zookeeper with modern Zookeeper version.
- Updates Curator
- Note: Exhibitor support is dropped in modern curator

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -319,7 +319,8 @@
         <slf4j.version>1.7.36</slf4j.version>
         <metrics.version>3.2.6</metrics.version>
         <mockito.version>4.11.0</mockito.version>
-        <zookeeper.version>3.5.9</zookeeper.version>
+        <zookeeper.version>3.8.2</zookeeper.version>
```
Comment (aishwaryasoni1991) [inline pom.xml:322]: The Zookeeper community has released 3.9.0. I get that 3.8.2 is still a stable release, but just getting the thoughts to adopt the latest .0 release.

File: storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java [hunk 2]
```diff
@@ -75,29 +72,7 @@ public List<ACL> getAclForPath(String s) {
...
-                        ObjectReader.getInt(conf.get(Config.STORM_EXHIBITOR_PORT)),
-                        new Exhibitors.BackupConnectionStringProvider() {
-                            @Override
-                            public String getBackupConnectionString() throws Exception {
-                                // use zk servers as backup if they exist
-                                return zkStr;
-                            }
-                        }),
-                new DefaultExhibitorRestClient(),
-                ObjectReader.getString(conf.get(Config.STORM_EXHIBITOR_URIPATH)),
```
Comment (agresch) [inline storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java:92]: are these config options no longer used?  Can we remove them as well?

File: storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java [hunk 3]
```diff
@@ -75,29 +72,7 @@ public List<ACL> getAclForPath(String s) {
...
-                ObjectReader.getInt(conf.get(Config.STORM_EXHIBITOR_POLL)),
-                new StormBoundedExponentialBackoffRetry(
```
Comment (agresch) [inline storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java:94]: I also see these in defaults.yaml.  Should they be removed?

storm.exhibitor.port: 8080
storm.exhibitor.poll.uripath: "/exhibitor/v1/cluster/list"

File: storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java (hunk 2 above)
Comment (rzo1) [reply storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java:92]: Good catch! Will clean it up tmrw.

**Review by bipinprasad — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (rzo1) [reply pom.xml]: I updated to 3.9.0 - let's see, if the build is ok with the change.

File: storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java (hunk 3 above)
Comment (rzo1) [reply storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java:94]: fixed

File: storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java (hunk 2 above)
Comment (rzo1) [reply storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java:92]: fixed

**Pull request merged**